File: introduction/ITP1_9_D-Transformation.java
import java.util.*;
public class Main {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    StringBuffer s = new StringBuffer(sc.next());
    int n = sc.nextInt();
    for(int i = 0; i < n; i++) {
      String o = sc.next();
      int a = sc.nextInt(), b = sc.nextInt()+1;
      String ss = s.substring(a, b);
      switch(o) {
        case "print":   System.out.println(ss); break;
        case "reverse": s = s.replace(a, b, new StringBuffer(ss).reverse().toString()); break;
        case "replace": s = s.replace(a, b, sc.next()); break;
      }
    }
  }
}

File: introduction/ITP1_9_D-Transformation.c
#include <stdio.h>
#include <string.h>
int main() {
  int n, i, j, a, b;
  char s[1001], ss[1001], o[7], p[1001];
  scanf("%s %d", s, &n);
  for(i=0; i<n; i++) {
    scanf("%s %d %d", o, &a, &b);
    if(!strcmp(o, "print")) {
      for(j=a;j<=b;j++) printf("%c", s[j]);
      printf("\n");
    }
    if(!strcmp(o, "reverse")) {
      strcpy(ss, s);
      for(j=a;j<=b;j++) s[j]=ss[b-(j-a)];
    }
    if(!strcmp(o, "replace")) {
      scanf("%s", p);
      for(j=a;j<=b;j++) s[j]=p[j-a];
    }
  }
  return 0;
}
